types/network.go

```diff
@@ -49,6 +49,9 @@ type ConnectionConfig struct {
 	// ExternalLabels specifies the external labels to be added to all samples
 	// sent to the Prometheus server.
 	ExternalLabels map[string]string
+	// Headers specifies the HTTP headers to be added to all requests
+	// sent to the server.
+	Headers map[string]string
 
```

Comment:
Technically should be something like map[string][]string but this is how remote write implements it.